projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,661 @@
+import {
+    AfterViewInit,
+    Component,
+    ContentChild,
+    ContentChildren,
+    ElementRef,
+    EventEmitter,
+    forwardRef,
+    HostBinding,
+    Input,
+    OnDestroy,
+    Output,
+    QueryList,
+    ViewChild,
+    TemplateRef
+} from '@angular/core';
+import { InteractionMode } from '../core/enums';
+import { IgxToggleDirective } from '../directives/toggle/toggle.directive';
+import { IgxCalendarComponent, WEEKDAYS } from '../calendar/index';
+import { AutoPositionStrategy, GlobalPositionStrategy, OverlaySettings } from '../services/index';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { IBaseEventArgs, KEYS, CancelableBrowserEventArgs, CancelableEventArgs } from '../core/utils';
+import { PositionSettings } from '../services/overlay/utilities';
+import { fadeIn, fadeOut } from '../animations/fade';
+import {
+    DateRange,
+    IgxDateEndComponent,
+    IgxDateSingleComponent,
+    IgxDateStartComponent,
+    IgxPickerToggleComponent,
+    IgxDateSeparatorDirective
+} from './igx-date-range-inputs.common';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { IToggleView } from '../core/navigation';
+import { IgxLabelDirective } from '../input-group';
+import { IgxInputGroupBase } from '../input-group/input-group.common';
+import { IgxDateTimeEditorEventArgs } from '../directives/date-time-editor';
+import { CurrentResourceStrings } from '../core/i18n/resources';
+
+
+// TODO: use default input format from date-time-editor.common?
+const DEFAULT_INPUT_FORMAT = 'dd/MM/yyyy';
+
+/**
+ * Range Date Picker provides the ability to select a range of dates from the calendar UI.
+ * It displays the range selection in a single or two input fields.
+ * @igxModule IgxDateRangeModule
+ *
+ * @igxTheme igx-input-group-theme, igx-calendar-theme
+ *
+ * @igxKeywords date, range, date range, date picker
+ *
+ * @igxGroup scheduling
+ *
+ * @remarks
+ * Range Date Picker provides the ability to select a range of dates from the calendar UI.
+ * It displays the range selection in a single or two input fields.
+ *
+ * @example:
+ * ```html
+ * <igx-date-range>
+ *  <input igxDateRangeStart>
+ *  <input igxDateRangeEnd>
+ * </igx-date-range>
+ * ```
+ * @example:
+ * ```html
+ * <igx-date-range mode="dropdown"></igx-date-range>
+ * ```
+ */
+@Component({
+    selector: 'igx-date-range',
+    templateUrl: './igx-date-range.component.html',
+    providers: [{ provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => IgxDateRangeComponent), multi: true }]
+})
+export class IgxDateRangeComponent implements IToggleView, AfterViewInit, OnDestroy, ControlValueAccessor {
```

Comment:
rename to `IgxDateRangePickerComponent`. Rename as well `igx-date-range-start`, `igx-date-range-end`, `igxDateRangeSeperator`